### PR TITLE
Adding CMK11

### PR DESCRIPTION
**Pull request by infinityis**
adding CMK11

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/custommk/cmk11/keymaps/default/rules.mk
Comment (zvecr) [inline keyboards/custommk/cmk11/keymaps/default/rules.mk:1]: This file should be removed.

File: keyboards/custommk/cmk11/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Copyright 2024 customMK
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_ESC,  KC_C,    KC_V,   KC_VOLU, KC_VOLD, KC_DEL,
+        KC_LCTL, KC_PGUP, KC_SPC, KC_PGDN, MO(1)
```
Comment (zvecr) [inline keyboards/custommk/cmk11/keymaps/default/keymap.c:9]: `MO(1)` here would go out of bounds.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/custommk/cmk11/keymaps/default/keymap.c (hunk 1 above)
Comment (infinityis) [reply keyboards/custommk/cmk11/keymaps/default/keymap.c]: Great catch! Thank you, I've fixed it.

File: keyboards/custommk/cmk11/keymaps/default/rules.mk
Comment (infinityis) [reply keyboards/custommk/cmk11/keymaps/default/rules.mk:1]: Thank you, done!

File: keyboards/custommk/cmk11/mcuconf.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// Used for audio
+#undef STM32_PWM_USE_TIM4
+#define STM32_PWM_USE_TIM4 TRUE
+
+// Used for FRAM
+#undef STM32_SPI_USE_SPI1
+#define STM32_SPI_USE_SPI1 TRUE
+
+// Used for RGB matrix
+#undef STM32_PWM_USE_TIM1 //timer 1 channel 3
```
Comment (drashna) [inline keyboards/custommk/cmk11/mcuconf.h:30]: Nitpick:  comment isn't needed.
```suggestion
#undef STM32_PWM_USE_TIM1
```

**Review by drashna — APPROVED** [review]

File: keyboards/custommk/cmk11/mcuconf.h (hunk 2 above)
Comment (infinityis) [reply keyboards/custommk/cmk11/mcuconf.h]: Comment has been removed. Thank you!

**Pull request merged**